src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/BlobManager.java

```diff
@@ -61,10 +65,24 @@ public BlobManager(
             if (cloudBlockBlob.getProperties().getLeaseStatus() == LeaseStatus.LOCKED) {
                 log.info(LEASE_ALREADY_ACQUIRED_MESSAGE, zipFilename, containerName);
                 return Optional.empty();
+            } else if (!readyToAcquireLease(cloudBlockBlob)) {
+                log.info(
+                    "Can't acquire lease on file {} in container {} "
+                        + "because lease was acquired less than {} seconds ago.",
+                    zipFilename,
+                    containerName,
+                    properties.getBlobLeaseAcquireDelayInSeconds()
+                );
+                return Optional.empty();
             }
 
             String leaseId = cloudBlockBlob.acquireLease(properties.getBlobLeaseTimeout(), null);
             log.info("Acquired lease on file {} in container {}. Lease ID: {}", zipFilename, containerName, leaseId);
+            // add lease acquired time to the blob metadata
+            cloudBlockBlob.getMetadata().put(
+                LEASE_ACQUIRED_TIME, LocalDateTime.now(EUROPE_LONDON_ZONE_ID).toString()
```

Comment:
might be better to send when it is going to expire?

Reply:
Updated to set the lease expire time